flowpipe/plug.py

```diff
@@ -298,54 +324,30 @@ def __init__(self, key, node, parent_plug, value=None):
             node (INode): The Node holding the Plug.
             parent_plug (InputPlug): The parent plug holding this Plug.
         """
-        super(SubInputPlug, self).__init__(
-            '{0}.{1}'.format(parent_plug.name, key), node,
-            (OutputPlug, SubOutputPlug))
+        # super().__init__() refers to self.parent_plug, so need to set it here
```

Comment:
Do we need this?

Reply:
The standard pattern is to call `super()` as the first line in an `__init__()`. Here, we need to deviate from that pattern, so I though it'd be good to inform other developers of why we do. I know that I, personally, will be confused by this when re-reading the code in six months :-)